celery/backends/database/models.py

```diff
@@ -30,18 +30,36 @@ class Task(ResultModelBase):
     date_done = sa.Column(sa.DateTime, default=datetime.utcnow,
                           onupdate=datetime.utcnow, nullable=True)
     traceback = sa.Column(sa.Text, nullable=True)
+    name = sa.Column(sa.String(155))
+    args = sa.Column(sa.String(155))
+    kwargs = sa.Column(sa.String(155))
+    worker = sa.Column(sa.String(155))
+    retries = sa.Column(sa.Integer)
+    queue = sa.Column(sa.String(155))
 
     def __init__(self, task_id):
         self.task_id = task_id
 
     def to_dict(self):
-        return {
+        results = {
             'task_id': self.task_id,
             'status': self.status,
             'result': self.result,
             'traceback': self.traceback,
             'date_done': self.date_done,
         }
+        if self.app.conf.find_value_for_key('extended', 'result'):
```

Comment:
Seems like you are accessing the app where it is not available.
You should probably require it as an argument to `__init__()`.

Reply:
@adimux took over the implementation and has fixed this issue.